ttnn/cpp/ttnn/operations/ccl/line_all_gather/device/line_all_gather_op.hpp

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+#include "common/core_coord.h"
+#include "impl/buffers/buffer.hpp"
+#include "tensor/tensor.hpp"
+#include "tt_dnn/op_library/ccl/shared_with_host/hetergeneous_data_structs.hpp"
+#include "tt_metal/common/constants.hpp"
+#include "tt_metal/host_api.hpp"
+#include "tt_dnn/op_library/ccl/ccl_host_datastructures.hpp"
+#include "tt_dnn/op_library/ccl/ccl_common.hpp"
+
+#include "tt_dnn/op_library/run_operation.hpp"
+
+#include <optional>
+#include <vector>
+#include <algorithm>
+
+namespace ttnn {
+
+namespace utils {
+
+namespace all_gather_op {
+using tt::tt_metal::ccl::Topology;
+}; // namespace all_gather_op
+
+using tt::tt_metal::ccl::EriscDatamoverBuilder;
```

Comment:
please avoid usings in headers as much as possible. here what you did is exposed this symbol inside ttnn::utils namespace, which seems fairly large